### ISSUE TITLE
Node-API (`nApi` feature flag) not working on Apple Silicon / MacOS ARM M1

### ISSUE THREAD

**Issue by XiNiHa** [prisma/prisma#7387]
### Bug description

The preview feature `nApi` is not working on Apple Silicon devices.
By directly loading the `.node` binary using plain `require()`, I found that the binary isn't for ARM64.
Here's the log with personal info removed.
```
> const prismaEngine = require('./prisma/libquery_engine_napi-darwin.dylib.node')
Uncaught:
Error: dlopen(/.../prisma/libquery_engine_napi-darwin.dylib.node, 1): no suitable image found.  Did find:
        /.../prisma/libquery_engine_napi-darwin.dylib.node: mach-o, but wrong architecture
        /.../prisma/libquery_engine_napi-darwin.dylib.node: mach-o, but wrong architecture
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1167:18)
    at Module.load (node:internal/modules/cjs/loader:973:32)
    at Function.Module._load (node:internal/modules/cjs/loader:813:14)
    at Module.require (node:internal/modules/cjs/loader:997:19)
    at require (node:internal/modules/cjs/helpers:92:18) {
  code: 'ERR_DLOPEN_FAILED'
}
```

I believe that it's just not yet supported as it's currently in the preview stage. However, I'm opening this since I found no tracking issue for Apple Silicon support of `nApi` feature.

### How to reproduce

1. Run `prisma generate` to generate the client and the engine.
2. Load the client
3. Error occurs

### Prisma information

```
generator client {
  provider        = "prisma-client-js"
  output          = "."
  previewFeatures = ["nApi"]
}

...
```

### Environment & setup

- OS: macOS 11.2.3 on M1 Macbook Air
- Node.js version: v15.5.0

### Prisma Version

```
prisma               : 2.23.0
@prisma/client       : 2.23.0
Current platform     : darwin
Query Engine         : query-engine adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt adf5e8cba3daf12d456d911d72b6e9418681b28b (at ../../node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : adf5e8cba3daf12d456d911d72b6e9418681b28b
Studio               : 0.393.0
```

**Comment by pantharshit00** [comment]
We don't really have a native build for Apple Silicon. The current version binaries works via Rosetta.

This is somewhat a duplicate of https://github.com/prisma/prisma/issues/5245 but I think with Node API this needs more attention and we should spend time setting up the CI for building these binaries.

**Comment by matthewmueller** [comment]
Yep, this is dependent on #5245. Tests with N-API + M1 definitely required here too.

**Comment by janpio** [comment]
Until that happens we will exclude M1/Apple Silicon from using Node API libraries in CLI and Client. Follow the implementation of that here: https://github.com/prisma/prisma/issues/7431 This will ensure you can work with Prisma.

**Comment by janpio** [comment]
This is what using it in Prisma Client looks like:
```
$ ts-node ./script.ts
PrismaClientInitializationError2 [PrismaClientInitializationError]:
Invalid `prisma.user.deleteMany()` invocation:


  Unable to load NAPI Library from /Users/administrator/actions-runner/_work/e2e-tests/e2e-tests/generic/basic/node_modules/.prisma/client/libquery_engine_napi-darwin.dylib.node, Library may be corrupt
    at cb (/Users/administrator/actions-runner/_work/e2e-tests/e2e-tests/generic/basic/node_modules/@prisma/client/runtime/index.js:33796:17) {
  clientVersion: '2.25.0-dev.39',
  errorCode: undefined
}
```


This is the error message when trying to generate a Prisma Client:
```
administrator@44528 basic % yarn prisma generate
yarn run v1.22.10
warning ../../package.json: No license field
$ /Users/administrator/actions-runner/_work/e2e-tests/e2e-tests/generic/basic/node_modules/.bin/prisma generate
Prisma schema loaded from prisma/schema.prisma
Error: dlopen(/Users/administrator/actions-runner/_work/e2e-tests/e2e-tests/generic/basic/node_modules/prisma/libquery_engine_napi-darwin.dylib.node, 1): no suitable image found.
  Did find:
        /Users/administrator/actions-runner/_work/e2e-tests/e2e-tests/generic/basic/node_modules/prisma/libquery_engine_napi-darwin.dylib.node: mach-o, but wrong architecture
        /Users/administrator/actions-runner/_work/e2e-tests/e2e-tests/generic/basic/node_modules/prisma/libquery_engine_napi-darwin.dylib.node: mach-o, but wrong architecture
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by matthewmueller** [comment]
Closing in favor of #5245
